src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/intervener/IntervenersAboutToSubmitHandler.java

```diff
@@ -54,21 +59,43 @@ public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(Finrem
         log.info("selected operation choice {} for intervener {} for Case ID: {}",
             selectedOperationCode, caseData.getIntervenersList().getValueCode(), caseId);
         List<String> errors = new ArrayList<>();
-        switch (selectedOperationCode) {
-            case ADD_INTERVENER_ONE_CODE -> service.updateIntervenerDetails(caseData.getIntervenerOne(), errors, callbackRequest);
-            case ADD_INTERVENER_TWO_CODE -> service.updateIntervenerDetails(caseData.getIntervenerTwo(), errors, callbackRequest);
-            case ADD_INTERVENER_THREE_CODE -> service.updateIntervenerDetails(caseData.getIntervenerThree(), errors, callbackRequest);
-            case ADD_INTERVENER_FOUR_CODE -> service.updateIntervenerDetails(caseData.getIntervenerFour(), errors, callbackRequest);
-            case DEL_INTERVENER_ONE_CODE -> service.removeIntervenerDetails(caseData.getIntervenerOne(), errors, caseData, caseId);
-            case DEL_INTERVENER_TWO_CODE -> service.removeIntervenerDetails(caseData.getIntervenerTwo(), errors, caseData, caseId);
-            case DEL_INTERVENER_THREE_CODE -> service.removeIntervenerDetails(caseData.getIntervenerThree(), errors, caseData, caseId);
-            case DEL_INTERVENER_FOUR_CODE -> service.removeIntervenerDetails(caseData.getIntervenerFour(), errors, caseData, caseId);
-            default -> throw new IllegalArgumentException("Invalid option received for case " + caseId);
+
+        if (ADD_OPERATION_CODES.contains(selectedOperationCode)) {
+            IntervenerWrapper intervener = getIntervenerWrapper(caseData, selectedOperationCode);
+            if (isIntervenerPostCodeMissing(intervener)) {
+                errors.add("Postcode field is required for the intervener.");
+            } else {
+                service.updateIntervenerDetails(intervener, errors, callbackRequest);
+            }
+        } else if (DELETE_OPERATION_CODES.contains(selectedOperationCode)) {
+            IntervenerWrapper intervener = getIntervenerWrapper(caseData, selectedOperationCode);
```

Comment:
No need to duplicate this line with the one in the ADD_OPERATION_CODES block - you can move them to before the if statement

Reply:
[1a9514c](https://github.com/hmcts/finrem-case-orchestration-service/pull/1641/commits/1a9514c2a51c04312192384722a4f1d21a0316d7)